components/service/nimbus/src/main/java/mozilla/components/service/nimbus/Nimbus.kt

```diff
@@ -0,0 +1,199 @@
+         * to indicate "undetermined".
+         *
+         * This method approximates the API21 method [Locale.toLanguageTag].
+         *
+         * @return a locale string that supports custom injected locale/languages.
+         */
+        internal fun getLocaleTag(): String {
+            // Thanks to toLanguageTag() being introduced in API21, we could have
+            // simple returned `locale.toLanguageTag();` from this function. However
+            // what kind of languages the Android build supports is up to the manufacturer
+            // and our apps usually support translations for more rare languages, through
+            // our custom locale injector. For this reason, we can't use `toLanguageTag`
+            // and must try to replicate its logic ourselves.
+            val locale = Locale.getDefault()
+            val language = getLanguageFromLocale(locale)
+            val country = locale.country // Can be an empty string.
+
+            return when {
+                language.isEmpty() -> "und"
+                country.isEmpty() -> language
+                else -> "$language-$country"
+            }
+        }
+
+        /**
+         * Sometimes we want just the language for a locale, not the entire language
+         * tag. But Java's .getLanguage method is wrong. A reference to the deprecated
+         * ISO language codes and their mapping can be found in [Locale.toLanguageTag] docs.
+         *
+         * @param locale a [Locale] object to be stringified.
+         * @return a language string, such as "he" for the Hebrew locales.
+         */
+        internal fun getLanguageFromLocale(locale: Locale): String {
+            // `locale.language` can, but should never be, an empty string.
+            // Modernize certain language codes.
+            return when (val language = locale.language) {
+                "iw" -> "he"
+                "in" -> "id"
+                "ji" -> "yi"
+                else -> language
+            }
+        }
+    }
+
+    private val scope: CoroutineScope = CoroutineScope(Dispatchers.Cached)
+
+    private lateinit var experiments: Experiments
+    private lateinit var dataDir: File
+    private var onExperimentUpdated: ((List<EnrolledExperiment>) -> Unit)? = null
+
+    init {
+        // Set the name of the native library
+        System.setProperty("uniffi.component.nimbus.libraryOverride", "nimbus")
+    }
+
+    /**
+     * Initialize the Nimbus SDK library.
+     *
+     * This should only be initialized once by the application.
+     *
+     * @param context [Context] to access application and device parameters.  As we cannot enforce
+     * through the compiler that the context pass to the initialize function is a Application
+     * Context, there could potentially be a memory leak if the initializing application doesn't
+     * comply.
+     *
+     * @return Returns a [Job] so that off-thread initialization can be monitored and interacted
```

Comment:
Is this comment stale or a TODO (or am I missing some kotlin magic? :)

Reply:
It's stale, I was originally intending to return the Job from the coroutine so that it could be awaited in tests to get around some of the async weirdness.